lib/containers/common.pm

```diff
@@ -231,37 +204,24 @@ sub test_container_image {
 
     my $smoketest = qq[/bin/sh -c '/bin/uname -r; /bin/echo "Heartbeat from $image"; ps'];
 
-    if ($runtime =~ /buildah/) {
-        if (script_run("buildah images | grep '$image'") != 0) {
-            assert_script_run("buildah pull $image", timeout => 300);
-            assert_script_run("buildah inspect --format='{{.FromImage}}' $image | grep '$image'");
-        }
-        # Ignore stderr explicitly, script_output with serial_terminal captures that as well.
-        my $container = script_output("buildah from $image 2>/dev/null");
-        record_info 'Container', qq[Testing:\nContainer "$container" based on image "$image"];
-        assert_script_run("buildah run $container $smoketest");
-    } else {
-        # Pull the image if necessary
-        if (script_run("$runtime image inspect --format='{{.RepoTags}}' $image | grep '$image'") != 0) {
-            # At least on publiccloud, this image pull can take long and occasinally fails due to network issues
-            script_retry("$runtime pull $image", retry => 3, timeout => 420);
-            assert_script_run("$runtime image inspect --format='{{.RepoTags}}' $image | grep '$image'");
-        }
-        assert_script_run("$runtime container create --name testing $image $smoketest");
-        assert_script_run("$runtime container start 'testing'");
-        assert_script_run("$runtime wait 'testing'", 90);
-        assert_script_run("$runtime container logs 'testing' | tee '$logfile'");
-        assert_script_run("$runtime container rm 'testing'");
-        if (script_run("grep \"`uname -r`\" '$logfile'") != 0) {
-            upload_logs("$logfile");
-            die "Kernel smoke test failed for $image";
-        }
-        if (script_run("grep \"Heartbeat from $image\" '$logfile'") != 0) {
-            upload_logs("$logfile");
-            die "Heartbeat test failed for $image";
-        }
-        assert_script_run "rm -f $logfile";
+    $runtime->pull($image, timeout => 420);
+    $runtime->check_image_in_host_registry($image);
+    $runtime->create_container($image, 'testing', $smoketest);
```

Comment:
I would prefer that runtime implements the function with all the input parameters in `%args`, so this call would turn into:
`$runtime->create_container(image => $image,  name => 'testing', cmd => $smoketest);`